### PR TITLE
Fixes #74: Create a shared base for addons.

### LINKED ISSUE #74: Build web extension compatible addons from single file
And use the build system to stick 'em in their directory at build-time.

### PR DESCRIPTION
**Pull request by miketaylr**
WIP.

### CODE REVIEW

File: firefox/lib/addon.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as _ from "../../shared/helpers.js";
+
+const REPORTER_ID = "addon-reporter-firefox";
+
+_.createContextMenu();
```
Comment (miketaylr) [inline firefox/lib/addon.js]: Eh, I guess it doesn't make sense to not stick this in the shared code. Should probably just have a `main` method and pass in the REPORTER_ID into that.

File: firefox/addon.js [hunk 2]
```diff
@@ -2,5 +2,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-// add a class to hide the download link when the add-on is installed.
-document.querySelector('.js-Navbar-link').classList.add('is-hidden');
+import initAddon from "../shared/base.js";
```
Comment (miketaylr) [inline firefox/addon.js]: Um, `opera/content.js → firefox/addon.js` oops? It won't matter... but 😆

**Comment by miketaylr** [comment]
Mostly happy with this now.

TODO:

* [ ] write chrome webpack config, move all those bits.
* [ ] write chrome webpack config, move all those bits.
* [ ] write commands to build them
* [ ] write docs on how commands work

Bonus points to write a command to package up crx, but that seems less important.

File: shared/manifest.json [hunk 3]
```diff
@@ -34,7 +28,7 @@
   },
   "browser_action": {
     "default_icon": {
-      "19": "icon32.png",
```
Comment (miketaylr) [inline shared/manifest.json]: This has to have been a typo...

File: .eslintrc [hunk 4]
```diff
@@ -1,19 +1,16 @@
 {
   "env": {
     "browser": true,
+    "node": true,
     "es6": true
   },
-  "extends": "eslint:recommended",
+  "extends": [
+    "eslint:recommended",
+    "plugin:prettier/recommended"
```
Comment (miketaylr) [inline .eslintrc]: Eh, I guess I added prettier to the mix in the middle of this... 🤷‍♂️ 🤷‍♀️

File: shared/manifest.json (hunk 3 above)
Comment (vladikoff) [reply shared/manifest.json]: lol

**Comment by magsout** [comment]
@miketaylr it's on my todolist ;)

File: shared/manifest.json (hunk 3 above)
Comment (magsout) [reply shared/manifest.json]: <3

**Comment by magsout** [comment]
@miketaylr r+

**Comment by miketaylr** [comment]
thanks!

**Pull request merged**